File: ExcelDivider/src/test/java/ru/tsystems/divider/context/MessageSimulator.java
package ru.tsystems.divider.context;


import org.jetbrains.annotations.NotNull;
import ru.tsystems.divider.exceptions.PropertiesException;
import ru.tsystems.divider.utils.api.MessageWorker;

import java.util.HashMap;
import java.util.Map;

import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_FORMAT_READ_COMMENT;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_FORMAT_READ_DATE;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_MODIFICATOR_KEYS_PRE;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_SYMBOLS_SOURCE_ANOTHER;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_SYMBOLS_SOURCE_ANOTHER_TASKS;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_SYMBOLS_SOURCE_COMMENTS;
import static ru.tsystems.divider.utils.constants.PropertiesConstantsKt.PROPS_SYMBOLS_SOURCE_EMPLOYEE;

public class MessageSimulator implements MessageWorker, ContextSimmulator<String> {

    private Map<String, String> properties;

    public void reset() {
        properties = new HashMap<>();
        properties.put(PROPS_MODIFICATOR_KEYS_PRE, "AD-");
        properties.put(PROPS_FORMAT_READ_DATE, null);
        properties.put(PROPS_SYMBOLS_SOURCE_EMPLOYEE, ",");
        properties.put(PROPS_SYMBOLS_SOURCE_ANOTHER_TASKS, ",");
        properties.put(PROPS_SYMBOLS_SOURCE_COMMENTS, ";");
        properties.put(PROPS_SYMBOLS_SOURCE_ANOTHER, ";");
        properties.put(PROPS_FORMAT_READ_COMMENT, "date,author");
    }

    public void setValue(String key, String value){
        properties.put(key, value);
    }

    public String getValue(String key){
        return properties.get(key);
    }

    protected MessageSimulator() {
        reset();
    }


    private static MessageSimulator messageSimulator;

    protected static MessageSimulator getMessageWorker() {
        if (messageSimulator == null) {
            messageSimulator = new MessageSimulator();
        }
        return messageSimulator;
    }

    @NotNull
    @Override
    public String getObligatorySourceValue(@NotNull String sourceName) throws PropertiesException{
        if(properties.get(sourceName) == null) {
            throw new PropertiesException(sourceName);
        }
        return properties.get(sourceName);
    }

    @Override
    public String getSourceValue(String sourcePath, String sourceName) {
        return properties.get(sourcePath + sourceName);
    }

    @Override
    public String getSourceValue(String sourceName) {
        return properties.get(sourceName);
    }

}

File: ExcelDivider/src/test/java/ru/tsystems/divider/context/MessageSimulator.kt
package ru.tsystems.divider.context

import ru.tsystems.divider.exceptions.PropertiesException
import ru.tsystems.divider.utils.api.MessageWorker
import ru.tsystems.divider.utils.constants.PROPS_FORMAT_READ_COMMENT
import ru.tsystems.divider.utils.constants.PROPS_FORMAT_READ_DATE
import ru.tsystems.divider.utils.constants.PROPS_MODIFICATOR_KEYS_PRE
import ru.tsystems.divider.utils.constants.PROPS_SYMBOLS_SOURCE_ANOTHER
import ru.tsystems.divider.utils.constants.PROPS_SYMBOLS_SOURCE_ANOTHER_TASKS
import ru.tsystems.divider.utils.constants.PROPS_SYMBOLS_SOURCE_COMMENTS
import ru.tsystems.divider.utils.constants.PROPS_SYMBOLS_SOURCE_EMPLOYEE
import java.util.HashMap

object MessageSimulator : MessageWorker, ContextSimmulator<String> {

    private lateinit var properties: MutableMap<String, String?>

    override fun reset() {
        properties = HashMap()
        properties[PROPS_MODIFICATOR_KEYS_PRE] = "AD-"
        properties[PROPS_FORMAT_READ_DATE] = null
        properties[PROPS_SYMBOLS_SOURCE_EMPLOYEE] = ","
        properties[PROPS_SYMBOLS_SOURCE_ANOTHER_TASKS] = ","
        properties[PROPS_SYMBOLS_SOURCE_COMMENTS] = ";"
        properties[PROPS_SYMBOLS_SOURCE_ANOTHER] = ";"
        properties[PROPS_FORMAT_READ_COMMENT] = "date,author"
    }

    override fun setValue(key: String, value: String) {
        properties[key] = value
    }

    override fun getValue(key: String): String? {
        return properties[key]
    }

    init {
        reset()
    }

    override fun getObligatorySourceValue(sourceName: String): String {
        return properties[sourceName]?:throw PropertiesException(sourceName)
    }

    override fun getSourceValue(sourcePath: String, sourceName: String): String? {
        return properties[sourcePath + sourceName]
    }

    override fun getSourceValue(sourceName: String): String? {
        return properties[sourceName]
    }
}